autogen/agentchat/groupchat.py

```diff
@@ -200,6 +193,10 @@ def next_agent(self, agent: Agent, agents: Optional[List[Agent]] = None) -> Agen
         """Return the next agent in the list."""
         if agents is None:
             agents = self.agents
+        else:
+            intersection = set(agents).intersection(set(self.agents))
+            if not len(intersection):
+                raise UndefinedNextAgent()
```

Comment:
The user used to be able to pass in unrelated agents, not sure if that's the desired behavior